workflow/util/util.go

```diff
@@ -689,6 +689,7 @@ func RetryWorkflow(kubeClient kubernetes.Interface, hydrator hydrator.Interface,
 	}
 
 	// Iterate the previous nodes. If it was successful Pod carry it forward
+	deletedNodes := make(map[string]bool)
```

Comment:
this only happens on retry - is something wrong with retry code ?

Reply:
Retry is the only code that deletes nodes off a Workflow.